pkg/kubectl/cmd/run.go

```diff
@@ -141,9 +179,43 @@ func addRunFlags(cmd *cobra.Command) {
 	cmd.Flags().String("schedule", "", i18n.T("A schedule in the Cron format the job should be run with."))
 }
 
-func RunRun(f cmdutil.Factory, cmdIn io.Reader, cmdOut, cmdErr io.Writer, cmd *cobra.Command, args []string, argsLenAtDash int) error {
+func (o *RunOpts) Complete(f cmdutil.Factory, cmd *cobra.Command) error {
+	o.ArgsLenAtDash = cmd.ArgsLenAtDash()
+	o.DryRun = cmdutil.GetFlagBool(cmd, "dry-run")
+	o.Expose = cmdutil.GetFlagBool(cmd, "expose")
+	o.Generator = cmdutil.GetFlagString(cmd, "generator")
+	o.Image = cmdutil.GetFlagString(cmd, "image")
+	o.Interactive = cmdutil.GetFlagBool(cmd, "stdin")
+	o.LeaveStdinOpen = cmdutil.GetFlagBool(cmd, "leave-stdin-open")
+	o.Port = cmdutil.GetFlagString(cmd, "port")
+	o.Quiet = cmdutil.GetFlagBool(cmd, "quiet")
+	o.Record = cmdutil.GetRecordFlag(cmd)
+	o.Schedule = cmdutil.GetFlagString(cmd, "schedule")
+	o.TTY = cmdutil.GetFlagBool(cmd, "tty")
+
+	attachFlag := cmd.Flags().Lookup("attach")
+	o.Attach = cmdutil.GetFlagBool(cmd, "attach")
+	if !attachFlag.Changed && o.Interactive {
+		o.Attach = true
+	}
+
+	if o.DryRun {
+		o.PrintFlags.Complete("%s (dry run)")
+	}
+	printer, err := o.PrintFlags.ToPrinter()
+	if err != nil {
+		return err
+	}
+	o.PrintObj = func(obj runtime.Object) error {
+		return printer.PrintObj(obj, o.Out)
+	}
+
+	return nil
+}
+
+func (o *RunOpts) Run(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
 	// Let kubectl run follow rules for `--`, see #13004 issue
-	if len(args) == 0 || argsLenAtDash == 0 {
+	if len(args) == 0 || o.ArgsLenAtDash == 0 {
```

Comment:
Once this merges, we should clean this and move them to `Validate`, but in a separate PR.